### PR TITLE
Resolves #9 Implement the Parallel Gateway element.

### LINKED ISSUE #9: Implement the Parallel Gateway element
Parallel Gateway is used to synchronize multiple concurrent branches (merging behavior). On the
other hand, it is used to spawn new concurrent threads on parallel branches (branching behavior).

**Operational Semantics**
The Parallel Gateway consumes exactly one token from each incoming Sequence Flow and produces exactly one token at each outgoing Sequence Flow. If there are excess tokens at an incoming Sequence Flow, these tokens remain at this Sequence Flow after execution of the Gateway.

**Exception Issues**: The Parallel Gateway cannot throw any exception.

### PR DESCRIPTION
**Pull request by caleeli**

### CODE REVIEW

File: src/ProcessMaker/Nayra/Bpmn/ParallelGatewayTrait.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+     *
+     * @param FlowNodeInterface $target
+     * @param callable $condition
+     * @param bool $default
+     *
+     * @return $this
+     *
+     * @codeCoverageIgnore
+     */
+    protected function buildConditionedConnectionTo(FlowNodeInterface $target, callable $condition, $default=false)
```
Comment (danloa) [inline src/ProcessMaker/Nayra/Bpmn/ParallelGatewayTrait.php]: Having dead code is a code smell, avoid forcing  a class to implement methods they don't use

File: src/ProcessMaker/Nayra/Bpmn/ParallelGatewayTransition.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    use TransitionTrait;
+
+    /**
+     * The Parallel Gateway consumes exactly one token from each incoming Sequence Flow
+     * and produces exactly one token at each outgoing Sequence Flow.
+     * If there are excess tokens at an incoming Sequence Flow, these tokens remain at
+     * this Sequence Flow after execution of the Gateway.
+     *
+     */
+    protected function initParallelGatewayTransition()
```
[the remark this answers is not in the document]
Comment (caleeli) [reply src/ProcessMaker/Nayra/Bpmn/ParallelGatewayTransition.php]: Fixed

File: src/ProcessMaker/Nayra/Bpmn/ParallelGatewayTransition.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+     * @return bool
+     */
+    protected function hasAllRequiredTokens()
+    {
+        $incomingWithToken = $this->incoming()->find(function(Connection $flow){
+            return $flow->originState()->getTokens()->count()>0;
+        });
+        return $incomingWithToken->count() === $this->incoming()->count();
+    }
+}
```
[the remark this answers is not in the document]
Comment (caleeli) [reply src/ProcessMaker/Nayra/Bpmn/ParallelGatewayTransition.php]: Fixed

File: src/ProcessMaker/Nayra/Bpmn/ParallelGatewayTrait.php (hunk 1 above)
Comment (caleeli) [reply src/ProcessMaker/Nayra/Bpmn/ParallelGatewayTrait.php]: Fixed

**Pull request merged**